Sources/Soto/Extensions/STS/STS+CredentialProvider.swift

```diff
@@ -135,6 +141,83 @@ extension STS {
         }
     }
 
+    /// Credential Provider using `AssumeRoleWebIdentity` to provide credentials. Uses environment variables to setup request structure. Used
+    /// by Amazon EKS Clusters.
+    struct AssumeRoleWithWebIdentityTokenFileCredentialProvider: CredentialProvider {
+        let tokenPromise: EventLoopPromise<STS.AssumeRoleWithWebIdentityRequest>
+        let webIdentityProvider: AssumeRoleWithWebIdentityCredentialProvider
+
+        init?(region: Region, context: CredentialProviderFactory.Context) {
+            guard let tokenFile = Environment["AWS_WEB_IDENTITY_TOKEN_FILE"] else { return nil }
+            guard let roleArn = Environment["AWS_ROLE_ARN"] else { return nil }
+            let sessionName = Environment["AWS_ROLE_SESSION_NAME"]
+
+            let tokenPromise = context.eventLoop.makePromise(of: STS.AssumeRoleWithWebIdentityRequest.self)
+            _ = Self.loadTokenFile(tokenFile, on: context.eventLoop).map { token in
+                let request = STS.AssumeRoleWithWebIdentityRequest(
+                    roleArn: roleArn,
+                    roleSessionName: sessionName ?? UUID().uuidString,
+                    webIdentityToken: token
+                )
+                tokenPromise.succeed(request)
+            }
```

Comment:
I think you would need to load the token file every time you want to make a new `AssumeRoleWithWebIdentityRequest`. So the `loadTokenFile` call should be part of the `requestProvider`. The reason is that the token itself might expire. EKS will automatically update the projected file with a new token though. So as long as you always read the token from disk, you should always be fine.

Reply:
ok will fix